src/test/java/com/google/sps/SessionContextTest.java

```diff
@@ -35,24 +35,36 @@ public void getLoggedInUser_ReturnUser() {
         com.google.appengine.api.users.User googleUser = new com.google.appengine.api.users.User("email", "domain");
         when(userService.getCurrentUser()).thenReturn(googleUser);
         when(userRepository.getUser(googleUser)).thenReturn(testUser);
-
-        User resultUser = sessionContext.getLoggedInUser();
-
-        Assert.assertEquals(testUser, resultUser);
+        
+        String resultUserName = "";
+        try {
+            User resultUser = sessionContext.getLoggedInUser();
+            resultUserName = resultUser.getName();
+        } catch(Exception e) {
+            Assert.fail("Exception caught" + e);
+        }
+
+        Assert.assertEquals(testUser.getName(), resultUserName);
     }
 
+   @Test
+   public void getLoggedInUser_ReturnNull() {
+       User testUser = null;
+       com.google.appengine.api.users.User googleUser = null;
+       when(userService.getCurrentUser()).thenReturn(googleUser);
+       
+       String resultUserName = "";
+        try {
+            User resultUser = sessionContext.getLoggedInUser();
+            if(resultUser == null) {
```

Comment:
L58-L61 just need to check that the return is null right?
`Assert.asssertEquals(null, sessionContext.getLoggedInUser()`

Reply:
yea but it gives an error for null, comparing to a user

Reply:
Nvm got it to work

Reply:
done